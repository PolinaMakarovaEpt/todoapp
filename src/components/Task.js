import React, { Component } from 'react';
import './todolist.css';

class Task extends Component {

  constructor(props) {
    super(props);

    this.parentDeleteCallback = props.deleteCallback;
    this.parentToggleCallback = props.toggleCallback;
  }

  deleteTask(e) {
    this.parentDeleteCallback(this.props.task.id, this.props.task.title);
  }

  toggleCheck(e) {
    let task = {...this.props.task};

    task.checked = !task.checked;

    this.parentToggleCallback(task);
  }

  render() {

    return (
      <div>

        <div className={this.props.task.checked ? "checked" : "task"} onClick={this.toggleCheck.bind(this)}>
          {this.props.task.title}
        </div>

        <span className="delete" onClick={this.deleteTask.bind(this)}>x</span> <hr />

      </div>
    );
  }
}

export default Task;
