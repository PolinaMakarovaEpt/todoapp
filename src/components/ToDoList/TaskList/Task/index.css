.task {
    display: block;
    padding: 0.4rem 2rem 0.5rem 1rem;
    margin: 0%;
}

.checked {
    display: block;
    padding: 0.4rem 2rem 0.5rem 1rem;
    margin: 0%;
    color:rgb(189, 189, 189);
    text-decoration: line-through;
}

.delete {
    position: relative;
    float: right;
    padding: 0rem 0.5rem 0.2rem 0.5rem;
    top: -1.65rem;
    z-index: 2;
    font-size: 22px;;
    color: red;

}
.delete:hover {
    background-color: red;
    color: white;

}