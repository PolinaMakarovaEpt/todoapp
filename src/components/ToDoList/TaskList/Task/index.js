import React, { Component } from 'react';
import './index.css';

class Task extends Component {

  constructor(props) {
    super(props);

    this.handleDelete = props.onDelete;
    this.handleCheck = props.onCheck;
  }

  onDelete(e) {
    this.handleDelete(this.props.task.id);
  }

  onCheck(e) {
    let task = { ...this.props.task };

    task.checked = !task.checked;

    this.handleCheck(task);
  }

  render() {

    return (
      <div>

        <div className={this.props.task.checked ? "checked" : "task"} onClick={this.onCheck.bind(this)}>
          {this.props.task.title}
        </div>

        <span className="delete" onClick={this.onDelete.bind(this)}>x</span> <hr />

      </div>
    );
  }
}

export default Task;
