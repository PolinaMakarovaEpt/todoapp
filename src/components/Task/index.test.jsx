import React from "react";
import Task from "./index";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import { JestEnvironment } from "@jest/environment";

configure({ adapter: new Adapter() });

it("matches the snapshot", () => {
    const props = {
        onDelete: jest.fn(),
        onCheck: jest.fn(),
        task: {
            id: 1,
            title: "",
            checked: false
        }
    }
    const tree = shallow(<Task {
        ...props
    } />);
    expect(toJson(tree)).toMatchSnapshot();
});
