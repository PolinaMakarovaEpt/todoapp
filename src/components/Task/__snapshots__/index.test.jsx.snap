// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <div
    className="task"
    onClick={[Function]}
  />
  <span
    className="delete"
    onClick={[Function]}
  >
    x
  </span>
   
  <hr />
</div>
`;
